stdlib/src/utils/string_slice.mojo

```diff
@@ -1047,24 +1047,21 @@ struct StringSlice[mut: Bool, //, origin: Origin[mut]](
                 offset += b_len
             return length != 0
 
-    fn splitlines[
-        O: ImmutableOrigin, //
-    ](self: StringSlice[O], keepends: Bool = False) -> List[StringSlice[O]]:
+    fn splitlines(
+        self: StringSlice, keepends: Bool = False
+    ) -> List[StringSlice[__type_of(self).origin]]:
```

Comment:
**Suggestion** I think it's possible to read the `origin` type parameter from `self` directly:

```suggestion
    ) -> List[StringSlice[self.origin]]:
```

Reply:
true, an even simpler version `fn splitlines(self, keepends: Bool = False) -> List[Self]:`